src/libprojectM/ProjectM.hpp

```diff
@@ -27,12 +27,15 @@
 #include "fatal.h"
 
 #ifdef WIN32
+
+#ifdef _MSC_VER
 // libs required for win32
 #pragma comment(lib, "psapi.lib")
 #pragma comment(lib, "kernel32.lib")
 
 #pragma warning(disable : 4244)
 #pragma warning(disable : 4305)
```

Comment:
Same here, but truncation of data, e.g. converting double to float.

Reply:
With the comments above in mind, say the #pragma's are removed from ProjectM.hpp. Then, what's left is an #include of `windows.h` in a public header, without `WIN32_LEAN_AND_MEAN` and friends defined! The `windows.h` header is really heavy and pulls in a lot of stuff, most notably a definition of `max` which breaks the standard library `numeric_limits<T>::max`.

I checked the `ProjectM.hpp` header quickly, and I think the `windows.h` header can safely be removed, and ditto for `unistd.h` on Unix(like). Then this whole thing can just be dropped. Do you have any thoughts about that?

Reply:
Any OS-specific headers we can drop, especially windows.h, would be a huge win. I think most of it is just legacy code, someone added it for some reason and it simply was left there.

Also, the lib pragmas should not be required anymore. If the libraries still must be linked specifically, this can be done via CMake, which would also be a better solution as the dependencies will be passed downstream if an application links the static library.

Reply:
After a lot of back and forth, I was not able to come up with a CMake construct to link `psapi`. So if we want to remove the pragma's in this PR all at once, I think I might just move them to `dirent.c` for the time being, unless you can come up with a solution. Unfortunately, that means that the MinGW build stays broken (GCC ignores pragma comment lib).

Reply:
From what I can see is that `psapi.lib` is only required for the `dlopen()` wrapper, which is used in the "Native Preset Factory". The native presets are long defunct and, with the restructured project, do no longer compile.

So my take on that would be removing the whole native preset stuff (which never really worked properly) and get rid of the whole issue. That said, just keep it as-is (with the `#pragma comment`), and I'll swing the axe to remove the broken code.